### PR TITLE
WFSS bugfixes for 0.12

### LINKED ISSUE #2640: WFSS partial order calculated correctly but not tracked explicitly
Doh. Oversight bug, the GrismObject that's returned for catalog items has a member to flag for partial orders being contained on the detector, however, this needs to be made into a dict that tracks per order. partial_orders is only informational right now.

### LINKED ISSUE #2641: NRC_TSGRISM extract_height optional parameter not being honored
The extract_height optional parameter into NIRCAM TSGRISM is not actually being honored because of logic that bounds the extraction box. The team expects a default extraction box for all extractions, so this is sugar.

### LINKED ISSUE #2642: WFSS extracted slits in MultiSlitModel missing bounding boxes
Add them.

### PR DESCRIPTION
**Pull request by sosey**
closes #2640 
* Make GrismObject.partial_order a lookup dict on order and fix partial_order logic
  * further testing showed that the partial orders weren't quite being calculated correctly either, now they should be, test forthcoming in future PR

closes #2641
* NRC_TSGRISM extract_height honored, bounding box fixed. In the case that the extraction size specified extends below 0, the box is cut from 0->extract_height because symmetry can't be maintained.  

closes #2642 
* added bounding boxes for the output SlitModel

### CODE REVIEW

**Comment by sosey** [comment]
@hbushouse nadia is traveling, can you review this and the test pr I'm also going to tag you on?

File: CHANGES.rst [hunk 1]
```diff
@@ -15,6 +15,8 @@ assign_wcs
 
 - Fix minor bug in catalog.utl.get_object_info()[#2550]
 
+- Make GrismObject.partial_order a lookup dict on order and fix partial_order logic
+
```
Comment (hbushouse) [inline CHANGES.rst]: Include a reference to the PR number

File: CHANGES.rst [hunk 2]
```diff
@@ -81,6 +83,8 @@ extract_2d
 
 - WFSS modes updated to allow specific order extraction, updated wavelengthrange reference file delivered as part of these changes [#1801]
 
+- add bounding box to wfss output SlitModel
```
Comment (hbushouse) [inline CHANGES.rst]: Include PR number

File: CHANGES.rst [hunk 3]
```diff
@@ -211,6 +215,7 @@ transforms
 
 - NIRISS models updated to allow for negative filter wheel rotations [#1801]
 
+- make partial_order attribute of GrismObject as lookup dict on order
```
Comment (hbushouse) [inline CHANGES.rst]: Include PR number

File: CHANGES.rst [hunk 4]
```diff
@@ -364,6 +369,8 @@ extract_2d
 
 - Fixed type specification for optional grism mode inputs [#2467]
 
+- NRC_TSGRISM extract_height honored, bounding box fixed
```
Comment (hbushouse) [inline CHANGES.rst]: Include PR number

**Pull request merged**